google/datalab/utils/_http.py

```diff
@@ -52,7 +52,7 @@ def __init__(self, status, content):
         error = error['errors'][0]
       self.message += ': ' + error['message']
     except Exception:
-      lines = content.split('\n') if isinstance(content, basestring) else []
+      lines = content.decode('utf-8').split('\n') if isinstance(content, basestring) else []
```

Comment:
I think the content arrives in python3 as bytes (bytestring) and not as string. That was the original error given that `b'bytes'.split('\n')` is invalid in python3.

Reply:
I don't know if this is a good assumption to make here. It's probably better to make this resilient to both types of content. This will also make the testing situation a lot easier.

Reply:
Right.. Let me have a think and do a bit of digging into python core and see if there's a better way to handle this..

Reply:
Thanks for working on this @alienczf!

Reply:
added a small workaround to split first and then cast.. it's still not watertight in terms of handling 'python2 bytes'